src/main/java/vazkii/botania/common/crafting/StateIngredientHelper.java

```diff
@@ -91,6 +133,42 @@ public static StateIngredient read(PacketBuffer buffer) {
 		}
 	}
 
+	/**
+	 * Resolves tag ingredients, returning null if their tag doesn't exist, then returns a new ingredient
+	 * if the replacement function returns a replacement, used to filter ores deprioritized with the config.
+	 */
+	@Nullable
+	public static StateIngredient resolveAndFilter(StateIngredient ingredient, @Nonnull UnaryOperator<List<Block>> function) {
+		if (ingredient instanceof StateIngredientTag) {
+			ITag<Block> tag = ((StateIngredientTag) ingredient).resolve();
+			if (tag == null) {
+				return null;
+			}
+			List<Block> blocks = tag.getAllElements();
+			if (blocks.isEmpty()) {
+				return null;
+			}
+			return getFiltered(ingredient, function, blocks);
+
+		} else if (ingredient instanceof StateIngredientBlocks) {
+			Collection<Block> blocks = ((StateIngredientBlocks) ingredient).getBlocks();
+			if (!(blocks instanceof List)) {
+				return ingredient;
```

Comment:
why?

Reply:
brain must have been tired at that point